### ISSUE TITLE
`cache_dir_lock` in `main` is always `None`, so the lockfile never gets cleaned up

### ISSUE THREAD

**Issue by guppy0130** [ansible/ansible-lint#4030]
##### Summary

`cache_dir_lock` is initialized as `None` here, in the global scope I believe: https://github.com/ansible/ansible-lint/blob/7849f27453fe1d9c29a43c71cff8f5b666bd43b6/src/ansiblelint/__main__.py#L82

In `initialize_options`, a new, frame-only `cache_dir_lock` is created here: https://github.com/ansible/ansible-lint/blob/7849f27453fe1d9c29a43c71cff8f5b666bd43b6/src/ansiblelint/__main__.py#L148-L152

I think pylint was correct to indicate that the global name was redefined. This variable isn't marked as `global` and so the `cache_dir_lock` in `initialize_options` is local to `initialize_options` only.

The value of `cache_dir_lock` (the `global` one) is still `None` later on, so this conditional is always falsy and the lockfile therefore never gets cleaned up: https://github.com/ansible/ansible-lint/blob/7849f27453fe1d9c29a43c71cff8f5b666bd43b6/src/ansiblelint/__main__.py#L380-L382

I believe this behavior is introduced by https://github.com/ansible/ansible-lint/commit/c6bd9a7a92592a8a700b26d659cf9232bd9eaba5.

I'd love to PR to resolve this, but I don't know if you prefer the `global` approach or just returning the lock from `initialize_options`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 24.2.0 using ansible-core:2.16.3 ansible-compat:4.1.11 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

In one session

```console
$ rm -r ~/.cache/ansible-compat/*
$ watch -n 1 'ls ~/.cache/ansible-compat/**/.lock'  # the quotes are so the shell under watch will expand it, not your current shell
```

and in another session

```console
$ pushd /tmp
$ mkdir blah
$ vim main.yml  # see below for MCVE playbook, but it can really be any playbook
$ ansible-lint
```

watch the lockfile get created when `ansible-lint` is run, and then it shouldn't get cleaned up.

##### Desired Behavior

The `cache_dir_lock` that `main` uses is not `None`, either by the `global` keyword or `initialize_options` returning it to its caller, `main`; in which case, the lock can be correctly cleaned up after ansible-lint finishes.

##### Actual Behavior

Ansible-lint doesn't clean up the lockfile, because `cache_dir_lock` is `None` in `main`.

Please give some details of what is happening. Include a [minimum complete
verifiable example] with:

- minimized playbook to reproduce the error

```yaml
- name: "MCVE"
  gather_facts: false
  hosts: "localhost"
  tasks:
    - name: "Hello world"
      ansible.builtin.debug:
        msg: "hello world"
```

- the output of running ansible-lint including the command line used

```text
Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
Seems legit, a pull-requests would be welcomed.
